addons/l10n_ar_base/models/res_partner.py

```diff
@@ -0,0 +1,126 @@
+            if rec.l10n_ar_id_category_id.afip_code != 80:
+                country = rec.country_id
+                if country and country.code != 'AR':
+                    if rec.is_company:
+                        rec.l10n_ar_cuit = country.l10n_ar_cuit_juridica
+                    else:
+                        rec.l10n_ar_cuit = country.l10n_ar_cuit_fisica
+                continue
+            if rec.l10n_ar_id_category_id.afip_code == 80:
+                rec.l10n_ar_cuit = rec.l10n_ar_id_number
+
+    @api.constrains('l10n_ar_id_number', 'l10n_ar_id_category_id')
+    def check_vat(self):
+        """ Update the the vat field using the information we have from
+        l10n_ar_id_number and l10n_ar_id_category_id fields
+        """
+        for rec in self:
+            if rec.l10n_ar_id_number and rec.l10n_ar_id_category_id and \
+               rec.l10n_ar_id_category_id.afip_code == 80:
+                rec.vat = 'AR' + rec.l10n_ar_id_number
+
+    @api.constrains('l10n_ar_id_number', 'l10n_ar_id_category_id')
+    def check_id_number_unique(self):
+        """ Taking into account the company's general settings it will check
+        that if the identification number we are trying to use is already set
+        in another partner.
+        """
+        if not safe_eval(self.env['ir.config_parameter'].sudo().get_param(
+                "l10n_ar_base.unique_id_numbers", 'False')):
+            return True
+        for rec in self:
+            # We allow same number in related partners
+            related_partners = rec.search([
+                '|', ('id', 'parent_of', rec.id),
+                ('id', 'child_of', rec.id)])
+            same_id_numbers = rec.search([
+                ('l10n_ar_id_number', '=', rec.l10n_ar_id_number),
+                ('l10n_ar_id_category_id', '=', rec.l10n_ar_id_category_id.id),
+                ('id', 'not in', related_partners.ids),
+            ])
+            if same_id_numbers:
+                raise ValidationError(_(
+                    'Identification number must be unique per Identification'
+                    ' category!\nSame number is only allowed for partners with'
+                    ' parent/child relation\n\n Already using this'
+                    ' number: ') + ', '.join(same_id_numbers.mapped('name'))
+                )
+
+    @api.model
+    def _name_search(self, name, args=None, operator='ilike', limit=100,
+                     name_get_uid=None):
+        """ We add the functionality to found partner by identification number
+        """
+        if not args:
+            args = []
+        # We used only this operarators in order to do not break the search.
+        # For example: when searching like "Does not containt" in name field
+        if name and operator in ('ilike', 'like', '=', '=like', '=ilike'):
+            recs = self.search(
+                [('l10n_ar_id_number', operator, name)] + args, limit=limit)
```

Comment:
Do you really type the vat number in the many2one partner-field?  One click further and you have the search bar (or search view).

Reply:
Yes, a lot. Odoo does it with the vat we also do it with the l10n_ar_id_number because is the usual field in Argentina
Every day the user choose lot of partners and avoiding one click is a lot